linux/storage/swap.sls

```diff
@@ -5,7 +5,27 @@
 
 {%- if swap.enabled %}
 
-{%- if swap.engine == 'file' %}
+{%- if swap.engine == 'partition' %}
+
+linux_create_swap_partition_{{ swap.device }}:
+  cmd.run:
+  - name: 'mkswap {{ swap.device }}'
+  - creates: {{ swap.device }}
```

Comment:
Can you add check if swap is already there? See following
https://github.com/tcpcloud/salt-formula-linux/blob/master/linux/storage/mount.sls#L13